### PR TITLE
Fix closure memory issue in GC

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Fix memory leak caused by post_time.jsx component.

Results are compared between release-4.5 branch and this commit. 
Note: Numbers might vary little depending on posts in thread.

Stats are for one of the two threads i was using for switch and same is being used for all comparisons.

Normal channel switch is about `3.4mb allocation` with `2.7mb retained` on every switch
screen shot 2017-12-09 at 3 00 40 pm

With the PR almost `same memory is allocated` and about `415kb is retained`(Mostly less than this).

screen shot 2017-12-09 at 2 40 36 pm

screen shot 2017-12-09 at 2 55 52 pm

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

**Comment by hmhealey** [comment]
I've added Work in Progress as we were discussing fixing this issue in other places too

**Comment by sudheerDev** [comment]
@hmhealey Done

File: components/post_view/post_time.jsx [hunk 1]
```diff
@@ -43,26 +43,18 @@ export default class PostTime extends React.PureComponent {
...
-        window.addEventListener('resize', () => {
-            updateWindowDimensions(this);
-        });
     }
 
     componentWillUnmount() {
         clearInterval(this.intervalId);
-        window.removeEventListener('resize', () => {
-            updateWindowDimensions(this);
-        });
```
Comment (sudheerDev) [inline components/post_view/post_time.jsx]: This was added to trigger render. Going from mobile screensize to desktop resolution does not change timestamp to link.

Comment (sudheerDev) [reply components/post_view/post_time.jsx]: @hmhealey I added these listeners for resize. PR is up for review as well

File: utils/utils.jsx [hunk 2]
```diff
@@ -1228,6 +1228,13 @@ export function windowHeight() {
     return $(window).height();
 }
 
+export function getWindowDimentions() {
```
Comment (saturninoabril) [inline utils/utils.jsx]: Please change to `getWindowDimensions`

File: components/search_results_item.jsx [hunk 3]
```diff
@@ -70,19 +68,25 @@ export default class SearchResultsItem extends React.Component {
...
     }
 
     componentWillUnmount() {
-        window.removeEventListener('resize', this.onUpdateWindowDimensions);
+        window.removeEventListener('resize', this.setDimentions);
     }
 
-    onUpdateWindowDimensions = () => {
-        Utils.updateWindowDimensions(this);
+    setDimentions = () => {
```
Comment (saturninoabril) [inline components/search_results_item.jsx]: Please change to `setDimensions`

File: components/rhs_root_post.jsx [hunk 4]
```diff
@@ -127,9 +113,27 @@ export default class RhsRootPost extends React.Component {
...
 
+    componentDidMount() {
+        window.addEventListener('resize', this.setDimentions);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.setDimentions);
+    }
+
+    setDimentions = () => {
```
Comment (saturninoabril) [inline components/rhs_root_post.jsx]: Please change to `setDimensions`

File: components/rhs_comment.jsx [hunk 5]
```diff
@@ -138,9 +109,43 @@ export default class RhsComment extends React.Component {
...
 
+    componentDidMount() {
+        window.addEventListener('resize', this.setDimentions);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.setDimentions);
+    }
+
+    setDimentions = () => {
```
Comment (saturninoabril) [inline components/rhs_comment.jsx]: Please change to `setDimensions`

File: components/search_results_item.jsx (hunk 3 above)
Comment (sudheerDev) [reply components/search_results_item.jsx]: Oops. Thanks fixed at all instances

**Comment by sudheerDev** [comment]
i find an issue here with post_time not working on mobile view. Let me fix it before merging this

**Comment by sudheerDev** [comment]
Done fixed it should be good to go!. 👍

**Comment by saturninoabril** [comment]
@sudheerDev Please rebase and we're good to merge this. Thanks!

**Comment by sudheerDev** [comment]
@saturninoabril Rebased

**Comment by sudheerDev** [comment]
@saturninoabril Rebased and did profile again just to cross check. Looks like master has improved on memory allocations as well. Memory retained is even less now

screen shot 2017-12-22 at 5 34 49 pm

**Comment by saturninoabril** [comment]
@sudheerDev Awesome, thanks!

**Pull request merged**